File: apps/s-2-admin/src/api/workflow/WorkflowCreateInput.ts
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";
import { WorkflowStepCreateNestedManyWithoutWorkflowsInput } from "./WorkflowStepCreateNestedManyWithoutWorkflowsInput";
import { WorkflowsOnEventTypeCreateNestedManyWithoutWorkflowsInput } from "./WorkflowsOnEventTypeCreateNestedManyWithoutWorkflowsInput";

export type WorkflowCreateInput = {
  name: string;
  user: UserWhereUniqueInput;
  trigger: "BEFORE_EVENT" | "EVENT_CANCELLED" | "NEW_EVENT";
  time?: number | null;
  timeUnit?: "DAY" | "HOUR" | "MINUTE" | null;
  steps?: WorkflowStepCreateNestedManyWithoutWorkflowsInput;
  activeOn?: WorkflowsOnEventTypeCreateNestedManyWithoutWorkflowsInput;
};

File: apps/cactus-demo-dotnet/src/APIs/Workflow/Dtos/WorkflowCreateInput.cs
using CactusDemoDotnet.Core.Enums;

namespace CactusDemoDotnet.APIs.Dtos;

public class WorkflowCreateInput
{
    public int? Id { get; set; }

    public string Name { get; set; }

    public User User { get; set; }

    public TriggerEnum Trigger { get; set; }

    public int? Time { get; set; }

    public TimeUnitEnum? TimeUnit { get; set; }

    public List<WorkflowStep>? Steps { get; set; }

    public List<WorkflowsOnEventType>? ActiveOn { get; set; }
}
